legacy/ui/legacy/src/main/java/com/fsck/k9/activity/MessageList.kt

```diff
@@ -1437,6 +1437,10 @@ open class MessageList :
             } else {
                 account = null
             }
+        } else {
+            if (account == null && search.accountUuids.size == 1) {
+                account = accountManager.getAccount(search.accountUuids[0])
+            }
         }
 
         configureDrawer()
```

Comment:
What happen when we get to this point and the account is still null? Does the NPE just get thrown later?

Are there any assertions or log messages we could add here so that someone with a NPE could provide us more information?

Reply:
It will still crash, that was the old behavior. Ideally this code needs to be rewritten without these local states.

Reply:
Can we add some sort of log info that would help us in the case where it still crashes?

Reply:
Not really, we would need to monitor the changes to the account reference. But the class is complex.

Reply:
Ok, very well. If you have ideas on how we could better monitor this without going overboard I'm open to it, but let's go ahead and merge this.